src/main/java/com/google/opensesame/projects/ProjectEntity.java

```diff
@@ -0,0 +1,116 @@
+package com.google.opensesame.projects;
+
+import com.google.appengine.api.datastore.PropertyContainer;
+import com.google.opensesame.github.GitHubGetter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.kohsuke.github.GHRepository;
+import org.kohsuke.github.GitHub;
+
+/** The class used to interact with the project entities in data storage. */
+public class ProjectEntity {
+  public static final String REPOSITORY_ID_KEY = "repositoryId";
+  public static final String NUM_MENTORS_KEY = "numMentors";
+  public static final String NUM_INTERESTED_USERS_KEY = "numInterestedUsers";
+  public static final String MENTOR_IDS_KEY = "mentorIds";
+  public static final String INTERESTED_USER_IDS_KEY = "interestedUserIds";
+
+  /**
+   * Creates a new ProjectEntity from a GitHub repository name.
+   *
+   * @param repositoryName
+   * @return Returns the new ProjectEntity or null if the repository name is invalid.
+   * @throws IOException
+   */
+  public static ProjectEntity fromRepositoryName(String repositoryName) throws IOException {
```

Comment:
Just a Q: If GitHub repository names are not unique, how can we get a unique ID based soley on the name?

Reply:
Do you also need an organization name, or GitHub info from the logged-in user for this function?

Reply:
Answered over chat but will answer here so others can see. The repository name takes the form `user/repo` or `org/repo` and in the short term it is unique. However org names and repo names can change so we only treat it as unique for this short timeframe between when the user supplies the repo URL and when it is converted into the permanent repo ID.